src/internal/_xtake.js

```diff
@@ -15,7 +15,8 @@ module.exports = (function() {
       return _reduced(result);
     } else {
       this.n -= 1;
-      return this.xf['@@transducer/step'](result, input);
+      return this.n === 0 ? _reduced(this.xf['@@transducer/step'](result, input))
+                          : this.xf['@@transducer/step'](result, input);
     }
```

Comment:
> Is there any way to simplify this?

Depends what you mean by simplify, but right now we are overloading `n`, so to speak. For me it would be simpler/clearer to add something like `this.idx = 0;` in the constructor and use that to count up towards `this.n` (which I think would also make the first `this.n === 0` check clearer).

Another alternative:

```js
    this.n -= 1;
    return (
      this.n < 0   ? _reduced(result) :
      this.n === 0 ? _reduced(this.xf['@@transducer/step'](result, input)) :
                     this.xf['@@transducer/step'](result, input));
```